File: src/irrgarten/FuzzyPlayer.java
package irrgarten;

import java.util.ArrayList;

public class FuzzyPlayer extends Player {
    FuzzyPlayer(Player ohter){
        super(ohter);
    }
    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves) {

        Directions preferred = super.move(direction, validMoves);
        return Dice.nextStep(preferred, validMoves, this.getIntellignece());
    }
    @Override
    public float attack(){
        return this.sumWeapons() + Dice.intensity(this.getStrength());
    }
    @Override
    protected float defensiveEnergy(){
        return this.sumShields() + Dice.intensity(this.getIntellignece());

    }
    @Override
    public String toString() {
        return "Fuzzy"+super.toString();
    }
}

File: JUEGO-RUBY/irrgarten/FuzzyPlayer.rb
require_relative 'Player'
require_relative 'Dice'


module Irrgarten
    class FuzzyPlayer < Player 
        def initialize(number,intelligence,strenght)
            super(number,intelligence,strenght)
        end
        def move(direction,validMoves)
            preferred = super(direction, validMoves)
            return Dice.next_step(preferred, validMoves, self.intelligence)
        end
       # def move(direction,validMoves)
            #preferred = super(direction,validMoves)
        #end
        def attack 
            return self.sum_weapons + Dice.intensity(self.strenght)
        end
        def defensiveEnergy
            self.sum_shields + Dice.intensity(self.intelligence)
        end
        def to_s
            return "Fuzzy " + super
        end
    end
end
